internal/provider/data_source_http.go

```diff
@@ -55,6 +57,18 @@ your control should be treated as untrustworthy.`,
 				Computed:    true,
 			},
 
+			"ca_certificate": {
+				Description: "PEM-encoded root certificate bundle for TLS authentication",
+				Type:        types.StringType,
+				Optional:    true,
+			},
+
+			"insecure": {
```

Comment:
FYI: soon it should be possible to use validators from https://github.com/hashicorp/terraform-plugin-framework-validators to apply a `schemavalidator.ConflictsWith`, and ensure these 2 new attributes are not set at the same time.

Reply:
Technically, setting insecure and ca_cert_pem should not a problem (makes no sense, but not reason to forbid?)

Reply:
So, one principle of HashiCorp that I feel we should always try to support/sustain, is that we need to _nudge_ practitioners in the right direction.

If a configuration combination doesn't make sense, we should entrust our tools to enforce that, this way we are _nudging_ the practitioner to do the right thing.

If waiting for those validators is a problem, could you please just capture the requirement in an issue an link it here for context?

Thank a lot.

Reply:
Sure, I will open an Issue after merge

Reply:
The `schemavalidator.ConflictsWith` is available. Could this be added to either `ca_cert_pem` or `insecure`?

Either the following on `ca_cert_pem`:
```go
Validators: []tfsdk.AttributeValidator{
	schemavalidator.ConflictsWith(path.MatchRoot("insecure")),
},
```

Or the following on `insecure`:
```go 
Validators: []tfsdk.AttributeValidator{
	schemavalidator.ConflictsWith(path.MatchRoot("ca_cert_pem")),
},
```

Could you also add a test to verify that an error is raised when both are set?